frontend/src/components/cart-list/CartContent.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from "react";
+
+import trash from "../../assets/svgs/trash.svg";
+
+import Quantity from "./Quantity";
+import Price from "./Price";
+
+import Image from "../ui/Image";
+
+import { CartDetail } from "../../types";
+
+import { useOrderPayListContext } from "../../hooks/useOrderPayListContext";
+import { useProductForm } from "../../hooks/useProductForm";
+import useCartListMutationDelete from "../../quires/useCartListMutationDelete";
+
+type CartContentProps = {
+  cart: CartDetail;
+};
+
+export default function CartContent({ cart }: CartContentProps) {
+  const { product } = cart;
+  const { mutate } = useCartListMutationDelete();
+  const [isChecked, setIsChecked] = useState(false);
+  const { getProductTotalPrice, updateQuantity, quantity } = useProductForm();
+  const {
+    addProductToOrderPayList,
+    deleteProductFromOrderPayList,
+    updateTotalPrice,
+    updateProductInOrderPayList,
+  } = useOrderPayListContext();
+
+  useEffect(() => {
+    updateProductInOrderPayList(product, quantity);
+  }, [quantity]);
+
+  const handleToggleProductSelection: React.MouseEventHandler<
+    HTMLInputElement
+  > = (event) => {
+    const { checked } = event.target;
+    setIsChecked(!isChecked);
+
+    if (checked) {
+      addProductToOrderPayList(product, quantity);
+      updateTotalPrice(product.price, quantity);
+    }
+
+    if (!checked) {
+      deleteProductFromOrderPayList(product.id);
+      updateTotalPrice(-product.price, quantity);
+    }
```

Comment:
early return 을 사용해도 되겠네요!

```suggestion
    if (checked) {
      addProductToOrderPayList(product, quantity);
      updateTotalPrice(product.price, quantity);
      return
    }
    
    deleteProductFromOrderPayList(product.id);
    updateTotalPrice(-product.price, quantity);
```